pkg/manager/runtime.go

```diff
@@ -448,31 +448,44 @@ func (v *VirtletRuntimeService) UpdateRuntimeConfig(context.Context, *kubeapi.Up
 // UpdateContainerResources stores in domain on libvirt info about Cpuset
 // for container then looks for running emulator and tries to adjust its
 // current settings through cgroups
+// the config.ResourceUpdateInProgress provides a simple locking mechanism here instead of
+// adding a pod-containerID based map here to track containers being updated
 func (v *VirtletRuntimeService) UpdateContainerResources(ctx context.Context, req *kubeapi.UpdateContainerResourcesRequest) (*kubeapi.UpdateContainerResourcesResponse, error) {
 	glog.V(4).Infof("Update Container Resources : %v", req)
-	lcr := req.GetLinux()
-	if lcr == nil {
-		glog.Errorf("linuxContainerResource is not set in UpdateContainerResourceRequest")
-		return &kubeapi.UpdateContainerResourcesResponse{}, nil
-	}
-
-	lr := linuxContinerResourceToLinuxResource(lcr)
 
 	containerId := req.ContainerId
 	info, err := v.virtTool.ContainerInfo(containerId)
 	if err != nil {
 		return nil, err
 	}
 
+	// if an resource being updated currently, just ignore the call, kubelet will retry periodically on the request
+	if info.Config.ResourceUpdateInProgress == true {
+		glog.V(4).Infof("VM resource update in progress. Ignore the new request")
+		return &kubeapi.UpdateContainerResourcesResponse{}, nil
+	}
+
+	glog.V(4).Infof("Update VM config to indicate update resource is in progress")
+	v.virtTool.SetUpdateResourceUpdateInProgress(info.Id, true)
```

Comment:
If the runtime crashed after you set inProgress to true, will it always stuck in that state even after restart? It looks like we need to reset it to false during runtime start.

Reply:
that is true for this particular container. some logic will need here i agree.

Reply:
Added reset code during runtime process restarts.